### PR TITLE
tests: benchmarks: Test fast UART with different GD frequencies

### PR DESCRIPTION
**Pull request by nordic-bami**
Default GD frequency is 320MHz
Validate fast UART with 256, 128 and 64MHz

### CODE REVIEW

**Review by nordic-segl — APPROVED** [review]

**Review by nordic-piks — APPROVED** [review]

**Review by nordic-pikr — APPROVED** [review]

**Comment by nordic-bami** [comment]
Please review: @nrfconnect/ncs-co-build-system

File: tests/benchmarks/multicore/idle_uarte/Kconfig [hunk 1]
```diff
@@ -0,0 +1,11 @@
+#
+# Copyright (c) 2024 Nordic Semiconductor ASA
+#
+# SPDX-License-Identifier: LicenseRef-Nordic-5-Clause
+#
+
+config GLOBAL_DOMAIN_FREQUENCY_SETTING
+	int "Global Domain frequency, 0 - 320MHz (default), 1 - 256MHz, 2 - 128MHz, 3 - 64MHz"
+	default 0
```
Comment (nordicjm) [inline tests/benchmarks/multicore/idle_uarte/Kconfig:9]: this should be a choice, then another (hidden) symbol that has defaults based on what choice symbol is selected

Comment (nordic-bami) [reply tests/benchmarks/multicore/idle_uarte/Kconfig:9]: This is integer used directly in code to automate the test, choice will make this unnecessarily complicated.

Comment (nordicjm) [reply tests/benchmarks/multicore/idle_uarte/Kconfig:9]: How?

C code becomes simpler:
```
const struct nrf_clock_spec clk_specs_global_hsfll = {
		.frequency = MHZ(CONFIG_TEST_CLOCK_SPEED_MHZ),
};
```

Kconfig code:
```
choice TEST_CLOCK_SPEED_OPTION
        prompt "Operational speed"
        default TEST_CLOCK_SPEED_OPTION_320MHZ

config TEST_CLOCK_SPEED_OPTION_320MHZ
        int "320MHz"

config TEST_CLOCK_SPEED_OPTION_256MHZ
        int "256MHz"

config TEST_CLOCK_SPEED_OPTION_128MHZ
        int "128MHz"

config TEST_CLOCK_SPEED_OPTION_64MHZ
        int "64MHz"

endchoice

config TEST_CLOCK_SPEED_MHZ
        int
        default 320 if TEST_CLOCK_SPEED_OPTION_320MHZ
        default 256 if TEST_CLOCK_SPEED_OPTION_256MHZ
        default 128 if TEST_CLOCK_SPEED_OPTION_128MHZ
        default 64 if TEST_CLOCK_SPEED_OPTION_64MHZ
```

**Comment by nordic-bami** [comment]
updated, ping @nrfconnect/ncs-co-build-system

**Review by nordicjm — APPROVED** [review]

**Pull request merged**